app/platform/platform.c

```diff
@@ -928,15 +928,26 @@ int platform_flash_erase_sector( uint32_t sector_id )
   return flash_erase( sector_id ) == SPI_FLASH_RESULT_OK ? PLATFORM_OK : PLATFORM_ERR;
 }
 
-uint32_t platform_flash_mapped2phys (uint32_t mapped_addr)
+static uint32_t flash_map_meg_offset ()
 {
   uint32_t cache_ctrl = READ_PERI_REG(CACHE_FLASH_CTRL_REG);
   if (!(cache_ctrl & CACHE_FLASH_ACTIVE))
     return -1;
   bool b0 = (cache_ctrl & CACHE_FLASH_MAPPED0) ? 1 : 0;
   bool b1 = (cache_ctrl & CACHE_FLASH_MAPPED1) ? 1 : 0;
-  uint32_t meg = (b1 << 1) | b0;
-  return mapped_addr - INTERNAL_FLASH_MAPPED_ADDRESS + meg * 0x100000;
+  return ((b1 << 1) | b0) * 0x100000;
+}
+ 
+uint32_t platform_flash_mapped2phys (uint32_t mapped_addr)
+{
+ uint32_t meg = flash_map_meg_offset();
+ return (meg&1) ? -1 : mapped_addr - INTERNAL_FLASH_MAPPED_ADDRESS + meg ;
```

Comment:
`(meg <= 0)` a bit clearer as to what the checks here (and a few lines further down) do?

Reply:
The platform API already had the mapping one way.  I just added the inverse map using the same checks.